scripts/replay-tx.js

```diff
@@ -0,0 +1,52 @@
+require("colors");
+const { Command } = require("commander");
+const log = require("single-line-log").stdout;
+const dotenv = require("dotenv").config();
+const { Contract, Wallet, utils } = require("ethers");
+const { getContractInterface } = require("@eth-optimism/contracts");
+const { JsonRpcProvider, getDefaultProvider } = require("@ethersproject/providers");
+const program = new Command();
+
+program.requiredOption("-h, --hash <number>", "specify transaction hash");
+program.requiredOption("-n, --network <string>", "specify network");
+program.option("-k, --key <number>", "specify private key");
+program.parse(process.argv);
+const argOptions = program.opts();
+
+const mainnetProvider = getDefaultProvider(argOptions.network);
+
+const L1_MESSENGER_PROXY = "0xD1EC7d40CCd01EB7A305b94cBa8AB6D17f6a9eFE";
+
+const wallet = new Wallet(argOptions.key, mainnetProvider);
+const proxyL1Messenger = new Contract(L1_MESSENGER_PROXY, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
+
+const main = async () => {
+  const receipt = await mainnetProvider.getTransactionReceipt(argOptions.hash);
```

Comment:
The script currently works "on-demand" only by a given transaction hash. After a conversation with @smartcontracts  understand the requirements of #705 are to have a listener which replays transactions that have failed execution on L2 for one of two reasons:
- They ran out of gas on L2 - in which case we need to listen for the `FailedRelayedMessage` event and replay those
-  The `ExecutionManager` is in a `paused` state - during an upgrade for example

@smartcontracts was going to check with @karlfloersch where in our infrastructure this listener should live. Please add details to the ticket.